<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Selecci√≥n de genes</title>
    <meta charset="utf-8" />
    <meta name="author" content="Yalbi I. Balderas-Mart√≠nez" />
    <meta name="date" content="2021-08-11" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# <strong>Selecci√≥n de genes</strong>
## <strong>Bioconductor</strong> para datos transcript√≥micos de c√©lula √∫nica (<strong>scRNA-seq</strong>) ‚Äì <strong>CDSB2021</strong>
### Yalbi I. Balderas-Mart√≠nez
### 2021-08-11

---






class: inverse

.center[

&lt;a href="https://osca.bioconductor.org/"&gt;&lt;img src="https://raw.githubusercontent.com/Bioconductor/OrchestratingSingleCellAnalysis-release/master/images/cover.png" style="width: 30%"/&gt;&lt;/a&gt;

&lt;a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"&gt;&lt;img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /&gt;&lt;/a&gt;&lt;br /&gt;This work is licensed under a &lt;a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"&gt;Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License&lt;/a&gt;.


]

.footnote[Descarga los materiales con `usethis::use_course('comunidadbioinfo/cdsb2021')` o rev√≠salos en l√≠nea via [**comunidadbioinfo.github.io/cdsb2021**](http://comunidadbioinfo.github.io/cdsb2021).]

&lt;style type="text/css"&gt;
/* From https://github.com/yihui/xaringan/issues/147  */
.scroll-output {
  height: 80%;
  overflow-y: scroll;
}
/* https://stackoverflow.com/questions/50919104/horizontally-scrollable-output-on-xaringan-slides */
pre {
  max-width: 100%;
  overflow-x: scroll;
}
/* From https://github.com/yihui/xaringan/wiki/Font-Size */
.tiny{
  font-size: 40%
}
/* From https://github.com/yihui/xaringan/wiki/Title-slide */
.title-slide {
  background-image: url(https://raw.githubusercontent.com/Bioconductor/OrchestratingSingleCellAnalysis/master/images/Workflow.png);
  background-size: 33%;
  background-position: 0% 100%
}
&lt;/style&gt;

---

# Diapositivas de Peter Hickey

Ver las diapositivas originales [aqu√≠](https://docs.google.com/presentation/d/19J2FyjKlBQdAkku4Oa6UZ6SA-Y4P7AEKCRIbEQWA9ho/edit#slide=id.ga100bba375887aa_0)

Plantilla desarrollada por: [Leonardo Collado-Torres](http://lcolladotor.github.io/)

---

# Motivaci√≥n

Usualmente usamos datos scRNA-seq para caracterizar la heterogeneidad entre c√©lulas

Para hacer esto, usamos m√©todos como el clustering y la reducci√≥n de dimensionalidad

Esto involucra resumir las diferencias por gen en una sola medida de (dis)similitud entre un par de c√©lulas

**¬øCu√°les genes deber√≠amos usar para calcular esta medida de (dis)similitud?**

---

# Selecci√≥n de _features_ (genes)

La elecci√≥n de los _features_ tiene un mayor impacto en qu√© tan similares decidimos que son las c√©lulas 

* ‚ûï _Features_ que contienen informaci√≥n √∫til biol√≥gica 
* ‚ûñ _Features_ que contienen ruido aleatorio
* üëâ Efectos laterales al reducir la dimensionalidad de los datos 

Deseamos seleccionar los **genes altamente variables** (High Variable Genes **HVGs**). Genes con una variaci√≥n incrementada en comparaci√≥n con otros genes que est√°n siendo afectados por ruido t√©cnico u otra variaci√≥n biol√≥gica que no es de nuestro inter√©s.

---
class: inverse center middle

# Dataset ilustrativo: PBMC4k 10X sin filtrar

---

# Dataset ilustrativo: PBMC4k 10X sin filtrar


```r
library(BiocFileCache)
bfc &lt;- BiocFileCache()
raw.path &lt;- bfcrpath(bfc, file.path(
    "http://cf.10xgenomics.com/samples",
    "cell-exp/2.1.0/pbmc4k/pbmc4k_raw_gene_bc_matrices.tar.gz"
))
untar(raw.path, exdir = file.path(tempdir(), "pbmc4k"))
```

---

# Dataset ilustrativo: PBMC4k 10X sin filtrar


```r
library(DropletUtils)
library(Matrix)
fname &lt;- file.path(tempdir(), "pbmc4k/raw_gene_bc_matrices/GRCh38")
sce.pbmc &lt;- read10xCounts(fname, col.names = TRUE)
```

Dataset "C√©lulas mononucleares humanas de sangre perif√©rica" de 10X Genomics

Descripci√≥n [aqu√≠](https://osca.bioconductor.org/unfiltered-human-pbmcs-10x-genomics.html)

.footnote[Zheng, G. X. Y. et al. Massively parallel digital transcriptional profiling of single cells. Nat. Commun. 8, 14049 (2017)]

---

# Dataset ilustrativo: PBMC4k 10X sin filtrar


```r
# gene-annotation
library(scater)
rownames(sce.pbmc) &lt;- uniquifyFeatureNames(
    rowData(sce.pbmc)$ID, rowData(sce.pbmc)$Symbol
)
library(EnsDb.Hsapiens.v86)
location &lt;- mapIds(EnsDb.Hsapiens.v86,
    keys = rowData(sce.pbmc)$ID,
    column = "SEQNAME", keytype = "GENEID"
)
# cell-detection
set.seed(100)
e.out &lt;- emptyDrops(counts(sce.pbmc))
sce.pbmc &lt;- sce.pbmc[, which(e.out$FDR &lt;= 0.001)]
```

.footnote[Zheng, G. X. Y. et al. Massively parallel digital transcriptional profiling of single cells. Nat. Commun. 8, 14049 (2017)]

---

# Dataset ilustrativo: PBMC4k 10X sin filtrar


```r
# quality-control
stats &lt;- perCellQCMetrics(sce.pbmc,
    subsets = list(Mito = which(location == "MT"))
)
high.mito &lt;- isOutlier(stats$subsets_Mito_percent,
    type = "higher"
)
sce.pbmc &lt;- sce.pbmc[, !high.mito]
# normalization
library(scran)
set.seed(1000)
clusters &lt;- quickCluster(sce.pbmc)
sce.pbmc &lt;- computeSumFactors(sce.pbmc, cluster = clusters)
sce.pbmc &lt;- logNormCounts(sce.pbmc)
```

---

class: inverse center middle

# Preguntas de repaso

* ¬øC√≥mo determinamos cu√°les eran los genes mitocondriales?
* ¬øC√≥mo decidimos filtrar las c√©lulas
* ¬øPuedes explicar c√≥mo normalizamos los datos?

---

class: inverse center middle

# Dataset ilustrativo: 416B

---

# Dataset ilustrativo: 416B


```r
library(scRNAseq)
sce.416b &lt;- LunSpikeInData(which="416b") 
sce.416b$block &lt;- factor(sce.416b$block)
```

L√≠nea celular de c√©lulas mieloides progenitoras inmortalizadas de rat√≥n usando SmartSeq2
[https://osca.bioconductor.org/lun-416b-cell-line-smart-seq2.html](https://osca.bioconductor.org/lun-416b-cell-line-smart-seq2.html)

.footnote[Lun, A. T. L., Calero-Nieto, F. J., Haim-Vilmovsky, L., G√∂ttgens, B. &amp; Marioni, J. C. Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data. Genome Res. 27, 1795‚Äì1806 (2017))]

---

# Dataset ilustrativo: 416B


```r
# gene-annotation
library(AnnotationHub)
ens.mm.v97 &lt;- AnnotationHub()[["AH73905"]]
rowData(sce.416b)$ENSEMBL &lt;- rownames(sce.416b)
rowData(sce.416b)$SYMBOL &lt;- mapIds(ens.mm.v97, 
  keys=rownames(sce.416b),
  keytype="GENEID", column="SYMBOL")
rowData(sce.416b)$SEQNAME &lt;- mapIds(ens.mm.v97,
  keys=rownames(sce.416b),
  keytype="GENEID", column="SEQNAME")
library(scater)
rownames(sce.416b) &lt;- uniquifyFeatureNames(rowData(sce.416b)$ENSEMBL, 
  rowData(sce.416b)$SYMBOL)
```

.footnote[Lun, A. T. L., Calero-Nieto, F. J., Haim-Vilmovsky, L., G√∂ttgens, B. &amp; Marioni, J. C. Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data. Genome Res. 27, 1795‚Äì1806 (2017))]

---

# Dataset ilustrativo: 416B


```r
# quality-control
mito &lt;- which(rowData(sce.416b)$SEQNAME=="MT")
stats &lt;- perCellQCMetrics(sce.416b, subsets=list(Mt=mito))
qc &lt;- quickPerCellQC(stats,
  percent_subsets=c("subsets_Mt_percent", "altexps_ERCC_percent"),
  batch=sce.416b$block)
sce.416b &lt;- sce.416b[,!qc$discard]

# normalization
library(scran)
sce.416b &lt;- computeSumFactors(sce.416b)
sce.416b &lt;- logNormCounts(sce.416b)
```

.footnote[Lun, A. T. L., Calero-Nieto, F. J., Haim-Vilmovsky, L., G√∂ttgens, B. &amp; Marioni, J. C. Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data. Genome Res. 27, 1795‚Äì1806 (2017))]

---

class: inverse center middle

# Preguntas de repaso

* ¬øC√≥mo determinamos cu√°les eran los genes mitocondriales?
* ¬øC√≥mo decidimos filtrar las c√©lulas
* ¬øPuedes explicar c√≥mo normalizamos los datos?

---

class: inverse center middle

# Cuantificando la varianza por gen

---

# Varianza de los _log-counts_

El enfoque m√°s simple para cuantificar la variaci√≥n _per-feature_ es simplemente calcular la varianza de los _log-counts_

* ‚ûï Selecci√≥n del _feature_  basado en los _log-counts_ (que ser√°n usadas en los an√°lisis m√°s adelante)
* ‚ö†Ô∏è La transformaci√≥n log no logra la estabilizaci√≥n de la varianza perfecta, as√≠ que se requiere modelar la relaci√≥n de la varianza-media de los _features_.

---

# Varianza de los _log-counts_

### Enfoque simple:
1. Calcular la varianza de los _log-counts_ para cada gen (ignorando grupos experimentales)
2. Ordenar los genes del m√°s-al-menos variable

### Un enfoque m√°s sofisticado:
1. Calcular la varianza de los _log-counts_ para cada gen (ignorando grupos experimentales)
2. Modelar la relaci√≥n entre la media y la varianza de los _log-counts_ para estimar la variaci√≥n _t√©cnica_
3. Estimar la varianza _biol√≥gica_ sustrayendo la varianza _t√©cnica_ de la varianza total
4. Ordenar los genes de la variable de mayor-a-menor biol√≥gicamente

---

# Varianza de los _log-counts_


```r
library(scran)
dec.pbmc &lt;- modelGeneVar(sce.pbmc)
```

* ü§ì El supuesto es que a cualquier abundancia dada, la abundancia de los perfiles de expresi√≥n de la mayor√≠a de los genes est√°n dominados por el ruido aleatorio _t√©cnico_
* ü§ì Por lo consiguiente, una tendencia representa un estimado del ruido t√©cnico como una funci√≥n de la abundancia
* ü§ì Podemos entonces descomponer la varianza total de cada gen en un componente _t√©cnico_ y uno _biol√≥gico_ 
* ü§ì Genes con una gran varianza _biol√≥gica_ son considerados interesantes 

---

# Varianza de los _log-counts_


```r
# Visualizing the fit - relaci√≥n entre la media y la varianza
fit.pbmc &lt;- metadata(dec.pbmc)
plot(fit.pbmc$mean, fit.pbmc$var,
    xlab = "Mean of log-expression",
    ylab = "Variance of log-expression"
)
curve(fit.pbmc$trend(x), col = "dodgerblue", add = TRUE, lwd = 2)
```

---

# Varianza de los _log-counts_

![](07-seleccion_genes_files/figure-html/unnamed-chunk-11-1.png)&lt;!-- --&gt;

---

# Varianza de los _log-counts_


```r
# Ordering by largest 'biological' variation to
# identify most interesting genes
dec.pbmc[order(dec.pbmc$bio, decreasing = TRUE), ]
```

```
## DataFrame with 33694 rows and 6 columns
##              mean     total      tech       bio      p.value          FDR
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;
## LYZ       1.95605   5.05854  0.835343   4.22320 1.10534e-270 2.17409e-266
## S100A9    1.93416   4.53551  0.835439   3.70007 2.71036e-208 7.61572e-205
## S100A8    1.69961   4.41084  0.824342   3.58650 4.31570e-201 9.43173e-198
## HLA-DRA   2.09785   3.75174  0.831239   2.92050 5.93940e-132 4.86758e-129
## CD74      2.90176   3.36879  0.793188   2.57560 4.83929e-113 2.50484e-110
## ...           ...       ...       ...       ...          ...          ...
## TMSB4X    6.08142  0.441718  0.679215 -0.237497     0.992447            1
## PTMA      3.82978  0.486454  0.731275 -0.244821     0.990002            1
## HLA-B     4.50032  0.486130  0.739577 -0.253447     0.991376            1
## EIF1      3.23488  0.482869  0.768946 -0.286078     0.995135            1
## B2M       5.95196  0.314948  0.654228 -0.339280     0.999843            1
```

---

# Coeficiente de variaci√≥n de las cuentas

El coeficiente de variaci√≥n de las cuentas al cuadrado (CV&lt;sup&gt;2&lt;/sup&gt;) es una alternativa a la varianza de los _log-counts_

* üëâ Se calcula usando las cuentas en lugar de los _log-counts_

* ü§ì CV es el _ratio_ de la desviaci√≥n est√°ndar a la media y est√° muy relacionada con el par√°metro de _dispersi√≥n_ de la distribuci√≥n binomial negativa usada en edgeR y DESeq2

---

# Coeficiente de variaci√≥n


```r
dec.cv2.pbmc &lt;- modelGeneCV2(sce.pbmc)
```

* ü§ì Modela la relaci√≥n de la media de la varianza cuando se considera la relevancia de cada gen
* ü§ì Asume que la mayor√≠a de los genes contienen ruido aleatorio y que la tendencia captura la mayor√≠a de la variaci√≥n t√©cnica 
* ü§ì Genes con un gran CV&lt;sup&gt;2&lt;/sup&gt; que se desv√≠an fuertemente de la tendencia es probable que representen genes afectados por la estructura biol√≥gica 
* ü§ì Usa la tasa (en lugar de la diferencia) del CV&lt;sup&gt;2&lt;/sup&gt; a la tendencia

---

# Coeficiente de variaci√≥n


```r
# Visualizing the fit
fit.cv2.pbmc &lt;- metadata(dec.cv2.pbmc)
plot(fit.cv2.pbmc$mean, fit.cv2.pbmc$cv2,
    log = "xy"
)
curve(fit.cv2.pbmc$trend(x),
    col = "dodgerblue",
    add = TRUE, lwd = 2
)
```

---

# Coeficiente de variaci√≥n

![](07-seleccion_genes_files/figure-html/unnamed-chunk-15-1.png)&lt;!-- --&gt;

---

# Coeficiente de variaci√≥n


```r
# Ordering by largest CV2 to identify most interesting genes
dec.cv2.pbmc[order(dec.cv2.pbmc$ratio,
    decreasing = TRUE
), ]
```

```
## DataFrame with 33694 rows and 6 columns
##                 mean     total     trend     ratio   p.value       FDR
##            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## PPBP       2.2437397   132.364  0.803689   164.696         0         0
## PRTFDC1    0.0658743  3197.564 20.266829   157.773         0         0
## HIST1H2AC  1.3731487   175.035  1.176934   148.721         0         0
## FAM81B     0.0477082  3654.419 27.902078   130.973         0         0
## PF4        1.8333127   109.451  0.935484   116.999         0         0
## ...              ...       ...       ...       ...       ...       ...
## AC023491.2         0       NaN       Inf       NaN       NaN       NaN
## AC233755.2         0       NaN       Inf       NaN       NaN       NaN
## AC233755.1         0       NaN       Inf       NaN       NaN       NaN
## AC213203.1         0       NaN       Inf       NaN       NaN       NaN
## FAM231B            0       NaN       Inf       NaN       NaN       NaN
```

---

# La varianza de los _log-counts_ vs coeficiente de variaci√≥n

**Generalmente se usa la varianza de los _log-counts_** 

* Ambas son medidas efectivas para cuantificar la variaci√≥n en la expresi√≥n g√©nica 
* CV&lt;sup&gt;2&lt;/sup&gt; tiende a tener otorgar rangos altos a genes bajos niveles de expresi√≥n (HGVs) 
	- √âstos son dirigidos por una sobreregulaci√≥n en subpoblaciones raras 
  - Puede asignar un alto rango a genes que no son de nuestro inter√©s con varianza baja absoluta	
* La variaci√≥n descrita por el CV&lt;sup&gt;2&lt;/sup&gt; de las cuentas es menos relevante para los procedimientos que operan en los _log-counts_

---

class: inverse center middle

# Cuantificando el ruido t√©cnico

---

# Cuantificando el ruido t√©cnico

* Previamente, ajustamos una l√≠nea de tendencia a todos los genes end√≥genos y asumimos que la mayor√≠a de los genes no est√°n dominados por ruido t√©cnico

* En la pr√°ctica, todos los genes expresados tienen alg√∫n nivel de variabilidad biol√≥gica diferente de cero (e.g., transcriptional bursting)

* Esto sugiere que nuestros estimados de los componentes t√©cnicos estar√°n inflados probablemente

---

# Cuantificando el ruido t√©cnico

* üëâ Es mejor que pensemos estos estimados como una variaci√≥n _t√©cnica_ m√°s la variaci√≥n biol√≥gica que no es interesante

* ü§î Pero que tal si un grupo de genes a una abundancia particular es afectado por un proceso biol√≥gico?
E.g., fuerte sobre regulaci√≥n de genes espec√≠ficos de un tipo celular podr√≠an conllevar a un enriquecimiento de HVGs en abundancias altas. Esto inflar√≠a la tendencia y compromete la detecci√≥n de los genes relevantes

**¬øC√≥mo podemos evitar este problema?**

---

# ¬øC√≥mo podemos evitar este problema y mejor estimar la variaci√≥n t√©cnica?

Podemos revisar dos enfoques:
1. Cuando tenemos spike-ins
2. Cuando no tenemos spike-ins 

---

# En la presencia de spike-ins


```r
dec.spike.416b &lt;- modelGeneVarWithSpikes(sce.416b, 
  "ERCC")
# Ordering by most interesting genes for 
# inspection.
dec.spike.416b[order(dec.spike.416b$bio, 
  decreasing=TRUE),]
```

```
## DataFrame with 46604 rows and 6 columns
##               mean     total      tech       bio      p.value          FDR
##          &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;
## Lyz2       6.61097   13.8497   1.57131   12.2784 1.48993e-186 1.54156e-183
## Ccl9       6.67846   13.1869   1.50035   11.6866 2.21855e-185 2.19979e-182
## Top2a      5.81024   14.1787   2.54776   11.6310  3.80016e-65  1.13040e-62
## Cd200r3    4.83180   15.5613   4.22984   11.3314  9.46221e-24  6.08574e-22
## Ccnb2      5.97776   13.1393   2.30177   10.8375  3.68706e-69  1.20193e-66
## ...            ...       ...       ...       ...          ...          ...
## Rpl5-ps2   3.60625  0.612623   6.32853  -5.71590     0.999616     0.999726
## Gm11942    3.38768  0.798570   6.51473  -5.71616     0.999459     0.999726
## Gm12816    2.91276  0.838670   6.57364  -5.73497     0.999422     0.999726
## Gm13623    2.72844  0.708071   6.45448  -5.74641     0.999544     0.999726
## Rps12l1    3.15420  0.746615   6.59332  -5.84670     0.999522     0.999726
```

üëâ Ajusta la tendencia solo con los spike-ins (que deber√≠an estar afectados solamente por variaci√≥n t√©cnica)

---

# En la presencia de spike-ins


```r
plot(dec.spike.416b$mean, dec.spike.416b$total,
  xlab="Mean of log-expression",
  ylab="Variance of log-expression")
fit.spike.416b &lt;- metadata(dec.spike.416b)
points(fit.spike.416b$mean, fit.spike.416b$var, 
  col="red", pch=16)
curve(fit.spike.416b$trend(x), col="dodgerblue", 
  add=TRUE, lwd=2)
```

---

# En la presencia de spike-ins

![](07-seleccion_genes_files/figure-html/unnamed-chunk-19-1.png)&lt;!-- --&gt;

---

# En la ausencia de spike-ins


```r
set.seed(0010101)
dec.pois.pbmc &lt;- modelGeneVarByPoisson(sce.pbmc)
# Ordering by most interesting genes for inspection.
dec.pois.pbmc[order(dec.pois.pbmc$bio,decreasing=TRUE),]
```

```
## DataFrame with 33694 rows and 6 columns
##              mean     total      tech        bio   p.value       FDR
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## LYZ       1.95605   5.05854  0.631190    4.42735         0         0
## S100A9    1.93416   4.53551  0.635102    3.90040         0         0
## S100A8    1.69961   4.41084  0.671491    3.73935         0         0
## HLA-DRA   2.09785   3.75174  0.604448    3.14730         0         0
## CD74      2.90176   3.36879  0.444928    2.92386         0         0
## ...           ...       ...       ...        ...       ...       ...
## ATP5J    0.614027  0.454754  0.502701 -0.0479473  0.943718  1.000000
## NEDD8    0.841407  0.561295  0.609895 -0.0486003  0.907530  0.997441
## NDUFA1   0.860049  0.560222  0.616330 -0.0561074  0.935049  1.000000
## SAP18    0.763057  0.515764  0.578890 -0.0631264  0.965173  1.000000
## SUMO2    1.359514  0.619811  0.693247 -0.0734360  0.960991  1.000000
```

* üëâ Realiza algunas asunciones estad√≠sticas acerca del ruido
* ü§ì Las cuentas UMI t√≠picamente muestran una variaci√≥n cercana a Poisson si solo consideramos ruido t√©cnico de la preparaci√≥n de las librer√≠as y la secuenciaci√≥n 
* ‚ö†Ô∏è modelGeneVarByPoisson() realiza simulaciones, por lo que necesitamos ‚Äúajustar la ‚Äúsemilla‚Äù para obtener resultados reproducibles
* ü§ì modelGeneVarByPoisson() pueden tambi√©n simular una variaci√≥n binomial negativa (variaci√≥n de Poisson sobredispersada)

---

# En la ausencia de spike-ins


```r
plot(dec.pois.pbmc$mean, dec.pois.pbmc$total, 
  pch=16, xlab="Mean of log-expression",
  ylab="Variance of log-expression")
curve(metadata(dec.pois.pbmc)$trend(x),
  col="dodgerblue", add=TRUE)
```

ü§ì La l√≠nea de tendencia basada puramente en ruido t√©cnico tiende a producir componentes ‚Äúbiol√≥gicos‚Äù m√°s grandes por los genes altamente expresados, que frecuentemente incluyen los genes ‚Äúhouse-keeping‚Äù
ü§î Necesitas considerar si tales genes son ‚Äúinteresantes‚Äù o no en tu dataset de inter√©s

---

# En la ausencia de spike-ins

![](07-seleccion_genes_files/figure-html/unnamed-chunk-22-1.png)&lt;!-- --&gt;

___

La presentaci√≥n fue hecha con el paquete de R [**xaringan**](https://github.com/yihui/xaringan) y configurada con [**xaringanthemer**](https://github.com/gadenbuie/xaringanthemer).

Este curso est√° basado en el libro [**Orchestrating Single Cell Analysis with Bioconductor**](https://osca.bioconductor.org/) de [Aaron Lun](https://www.linkedin.com/in/aaron-lun-869b5894/), [Robert Amezquita](https://robertamezquita.github.io/), [Stephanie Hicks](https://www.stephaniehicks.com/) y [Raphael Gottardo](http://rglab.org), adem√°s del [**curso de scRNA-seq para WEHI**](https://drive.google.com/drive/folders/1cn5d-Ey7-kkMiex8-74qxvxtCQT6o72h) creado por [Peter Hickey](https://www.peterhickey.org/).

Puedes encontrar los archivos para este taller en  [comunidadbioinfo/cdsb2021](https://github.com/comunidadbioinfo/cdsb2021). 

Instructora: [**Yalbi I. Balderas-Mart√≠nez**](http://Yalbibalderas.github.io/).

.footnote[Descarga los materiales con `usethis::use_course('comunidadbioinfo/cdsb2021')` o rev√≠salos en l√≠nea v√≠a [**comunidadbioinfo.github.io/cdsb2021**](http://comunidadbioinfo.github.io/cdsb2021).]

---

# Detalles de la sesi√≥n de R


.scroll-output[
.tiny[


```r
options(width = 120)
sessioninfo::session_info()
```

```
## ‚îÄ Session info ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
##  setting  value                       
##  version  R version 4.0.5 (2021-03-31)
##  os       Ubuntu 20.10                
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2021-08-11                  
## 
## ‚îÄ Packages ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
##  package                * version    date       lib source                       
##  AnnotationDbi          * 1.52.0     2020-10-27 [1] Bioconductor                 
##  AnnotationFilter       * 1.14.0     2020-10-27 [2] Bioconductor                 
##  AnnotationHub          * 2.22.1     2021-04-16 [2] Bioconductor                 
##  askpass                  1.1        2019-01-13 [3] CRAN (R 4.0.1)               
##  assertthat               0.2.1      2019-03-21 [3] CRAN (R 4.0.1)               
##  beachmat                 2.6.4      2020-12-20 [2] Bioconductor                 
##  beeswarm                 0.4.0      2021-06-01 [2] CRAN (R 4.0.5)               
##  Biobase                * 2.50.0     2020-10-27 [1] Bioconductor                 
##  BiocFileCache          * 1.14.0     2020-10-27 [1] Bioconductor                 
##  BiocGenerics           * 0.36.0     2020-10-27 [1] Bioconductor                 
##  BiocManager              1.30.12    2021-03-28 [1] CRAN (R 4.0.5)               
##  BiocNeighbors            1.8.2      2020-12-07 [2] Bioconductor                 
##  BiocParallel             1.24.1     2020-11-06 [1] Bioconductor                 
##  BiocSingular             1.6.0      2020-10-27 [2] Bioconductor                 
##  BiocVersion              3.12.0     2020-04-27 [1] Bioconductor                 
##  biomaRt                  2.46.3     2021-02-09 [1] Bioconductor                 
##  Biostrings               2.58.0     2020-10-27 [1] Bioconductor                 
##  bit                      4.0.4      2020-08-04 [1] CRAN (R 4.0.5)               
##  bit64                    4.0.5      2020-08-30 [1] CRAN (R 4.0.5)               
##  bitops                   1.0-6      2013-08-17 [1] CRAN (R 4.0.5)               
##  blob                     1.2.2      2021-07-23 [3] CRAN (R 4.0.5)               
##  bluster                  1.0.0      2020-10-27 [2] Bioconductor                 
##  bslib                    0.2.5.1    2021-05-18 [3] CRAN (R 4.0.5)               
##  cachem                   1.0.4      2021-02-13 [1] CRAN (R 4.0.5)               
##  cli                      3.0.1      2021-07-17 [3] CRAN (R 4.0.5)               
##  colorspace               2.0-2      2021-06-24 [2] CRAN (R 4.0.5)               
##  crayon                   1.4.1      2021-02-08 [3] CRAN (R 4.0.5)               
##  curl                     4.3.2      2021-06-23 [3] CRAN (R 4.0.5)               
##  DBI                      1.1.1      2021-01-15 [3] CRAN (R 4.0.5)               
##  dbplyr                 * 2.1.1      2021-04-06 [3] CRAN (R 4.0.5)               
##  DelayedArray             0.16.3     2021-03-24 [1] Bioconductor                 
##  DelayedMatrixStats       1.12.3     2021-02-03 [2] Bioconductor                 
##  digest                   0.6.27     2020-10-24 [3] CRAN (R 4.0.5)               
##  dplyr                    1.0.7      2021-06-18 [2] CRAN (R 4.0.5)               
##  dqrng                    0.3.0      2021-05-01 [2] CRAN (R 4.0.5)               
##  DropletUtils           * 1.10.3     2021-02-02 [2] Bioconductor                 
##  edgeR                    3.32.1     2021-01-14 [2] Bioconductor                 
##  ellipsis                 0.3.2      2021-04-29 [3] CRAN (R 4.0.5)               
##  EnsDb.Hsapiens.v86     * 2.99.0     2021-08-08 [2] Bioconductor                 
##  ensembldb              * 2.14.1     2021-04-19 [2] Bioconductor                 
##  evaluate                 0.14       2019-05-28 [3] CRAN (R 4.0.1)               
##  ExperimentHub            1.16.1     2021-04-16 [2] Bioconductor                 
##  fansi                    0.5.0      2021-05-25 [2] CRAN (R 4.0.5)               
##  fastmap                  1.1.0      2021-01-25 [3] CRAN (R 4.0.5)               
##  generics                 0.1.0      2020-10-31 [3] CRAN (R 4.0.5)               
##  GenomeInfoDb           * 1.26.4     2021-03-10 [1] Bioconductor                 
##  GenomeInfoDbData         1.2.4      2021-04-05 [1] Bioconductor                 
##  GenomicAlignments        1.26.0     2020-10-27 [1] Bioconductor                 
##  GenomicFeatures        * 1.42.3     2021-04-01 [1] Bioconductor                 
##  GenomicRanges          * 1.42.0     2020-10-27 [1] Bioconductor                 
##  ggbeeswarm               0.6.0      2017-08-07 [2] CRAN (R 4.0.5)               
##  ggplot2                * 3.3.3      2020-12-30 [1] CRAN (R 4.0.5)               
##  glue                     1.4.2      2020-08-27 [3] CRAN (R 4.0.5)               
##  gridExtra                2.3        2017-09-09 [3] CRAN (R 4.0.1)               
##  gtable                   0.3.0      2019-03-25 [3] CRAN (R 4.0.1)               
##  HDF5Array                1.18.1     2021-02-04 [2] Bioconductor                 
##  highr                    0.9        2021-04-16 [3] CRAN (R 4.0.5)               
##  hms                      1.1.0      2021-05-17 [3] CRAN (R 4.0.5)               
##  htmltools                0.5.1.1    2021-01-22 [3] CRAN (R 4.0.5)               
##  httpuv                   1.6.1      2021-05-07 [3] CRAN (R 4.0.5)               
##  httr                     1.4.2      2020-07-20 [3] CRAN (R 4.0.5)               
##  igraph                   1.2.6      2020-10-06 [2] CRAN (R 4.0.5)               
##  interactiveDisplayBase   1.28.0     2020-10-27 [2] Bioconductor                 
##  IRanges                * 2.24.1     2020-12-12 [1] Bioconductor                 
##  irlba                    2.3.3      2019-02-05 [1] CRAN (R 4.0.5)               
##  jquerylib                0.1.4      2021-04-26 [3] CRAN (R 4.0.5)               
##  jsonlite                 1.7.2      2020-12-09 [3] CRAN (R 4.0.5)               
##  knitr                    1.33       2021-04-24 [2] CRAN (R 4.0.5)               
##  later                    1.2.0      2021-04-23 [3] CRAN (R 4.0.5)               
##  lattice                  0.20-44    2021-05-02 [4] CRAN (R 4.0.5)               
##  lazyeval                 0.2.2      2019-03-15 [2] CRAN (R 4.0.5)               
##  lifecycle                1.0.0      2021-02-15 [3] CRAN (R 4.0.5)               
##  limma                    3.46.0     2020-10-27 [1] Bioconductor                 
##  locfit                   1.5-9.4    2020-03-25 [1] CRAN (R 4.0.5)               
##  magrittr                 2.0.1      2020-11-17 [3] CRAN (R 4.0.5)               
##  Matrix                 * 1.3-4      2021-06-01 [4] CRAN (R 4.0.5)               
##  MatrixGenerics         * 1.2.1      2021-01-30 [1] Bioconductor                 
##  matrixStats            * 0.58.0     2021-01-29 [1] CRAN (R 4.0.5)               
##  memoise                  2.0.0      2021-01-26 [1] CRAN (R 4.0.5)               
##  mime                     0.11       2021-06-23 [3] CRAN (R 4.0.5)               
##  munsell                  0.5.0      2018-06-12 [3] CRAN (R 4.0.1)               
##  openssl                  1.4.4      2021-04-30 [3] CRAN (R 4.0.5)               
##  pillar                   1.6.2      2021-07-29 [3] CRAN (R 4.0.5)               
##  pkgconfig                2.0.3      2019-09-22 [3] CRAN (R 4.0.1)               
##  prettyunits              1.1.1      2020-01-24 [3] CRAN (R 4.0.1)               
##  progress                 1.2.2      2019-05-16 [3] CRAN (R 4.0.1)               
##  promises                 1.2.0.1    2021-02-11 [3] CRAN (R 4.0.5)               
##  ProtGenerics             1.22.0     2020-10-27 [2] Bioconductor                 
##  purrr                    0.3.4      2020-04-17 [3] CRAN (R 4.0.1)               
##  R.methodsS3              1.8.1      2020-08-26 [1] CRAN (R 4.0.5)               
##  R.oo                     1.24.0     2020-08-26 [1] CRAN (R 4.0.5)               
##  R.utils                  2.10.1     2020-08-26 [1] CRAN (R 4.0.5)               
##  R6                       2.5.0      2020-10-28 [3] CRAN (R 4.0.5)               
##  rappdirs                 0.3.3      2021-01-31 [1] CRAN (R 4.0.5)               
##  Rcpp                     1.0.7      2021-07-07 [3] CRAN (R 4.0.5)               
##  RCurl                    1.98-1.3   2021-03-16 [1] CRAN (R 4.0.5)               
##  rhdf5                    2.34.0     2020-10-27 [2] Bioconductor                 
##  rhdf5filters             1.2.1      2021-05-03 [2] Bioconductor                 
##  Rhdf5lib                 1.12.1     2021-01-26 [2] Bioconductor                 
##  rlang                    0.4.10     2020-12-30 [1] CRAN (R 4.0.5)               
##  rmarkdown                2.10       2021-08-06 [2] CRAN (R 4.0.5)               
##  Rsamtools                2.6.0      2020-10-27 [1] Bioconductor                 
##  RSQLite                  2.2.5      2021-03-27 [1] CRAN (R 4.0.5)               
##  rstudioapi               0.13       2020-11-12 [3] CRAN (R 4.0.5)               
##  rsvd                     1.0.5      2021-04-16 [2] CRAN (R 4.0.5)               
##  rtracklayer              1.50.0     2020-10-27 [1] Bioconductor                 
##  S4Vectors              * 0.28.1     2020-12-09 [1] Bioconductor                 
##  sass                     0.4.0.9000 2021-08-09 [2] Github (rstudio/sass@f148415)
##  scales                   1.1.1      2020-05-11 [3] CRAN (R 4.0.1)               
##  scater                 * 1.18.6     2021-02-26 [2] Bioconductor                 
##  scran                  * 1.18.7     2021-04-16 [2] Bioconductor                 
##  scRNAseq               * 2.4.0      2020-11-09 [2] Bioconductor                 
##  scuttle                  1.0.4      2020-12-17 [2] Bioconductor                 
##  sessioninfo              1.1.1      2018-11-05 [2] CRAN (R 4.0.5)               
##  shiny                    1.6.0      2021-01-25 [3] CRAN (R 4.0.5)               
##  SingleCellExperiment   * 1.12.0     2020-10-27 [2] Bioconductor                 
##  sparseMatrixStats        1.2.1      2021-02-02 [2] Bioconductor                 
##  statmod                  1.4.35     2020-10-19 [1] CRAN (R 4.0.5)               
##  stringi                  1.7.3      2021-07-16 [3] CRAN (R 4.0.5)               
##  stringr                  1.4.0      2019-02-10 [3] CRAN (R 4.0.1)               
##  SummarizedExperiment   * 1.20.0     2020-10-27 [1] Bioconductor                 
##  tibble                   3.1.3      2021-07-23 [3] CRAN (R 4.0.5)               
##  tidyselect               1.1.1      2021-04-30 [3] CRAN (R 4.0.5)               
##  utf8                     1.2.2      2021-07-24 [3] CRAN (R 4.0.5)               
##  vctrs                    0.3.8      2021-04-29 [3] CRAN (R 4.0.5)               
##  vipor                    0.4.5      2017-03-22 [2] CRAN (R 4.0.5)               
##  viridis                  0.5.1      2018-03-29 [1] CRAN (R 4.0.5)               
##  viridisLite              0.4.0      2021-04-13 [3] CRAN (R 4.0.5)               
##  whisker                  0.4        2019-08-28 [3] CRAN (R 4.0.1)               
##  withr                    2.4.2      2021-04-18 [3] CRAN (R 4.0.5)               
##  xaringan                 0.22       2021-06-23 [1] CRAN (R 4.0.5)               
##  xaringanthemer         * 0.4.0      2021-06-24 [2] CRAN (R 4.0.5)               
##  xfun                     0.25       2021-08-06 [3] CRAN (R 4.0.5)               
##  XML                      3.99-0.6   2021-03-16 [1] CRAN (R 4.0.5)               
##  xml2                     1.3.2      2020-04-23 [3] CRAN (R 4.0.1)               
##  xtable                   1.8-4      2019-04-21 [3] CRAN (R 4.0.1)               
##  XVector                  0.30.0     2020-10-27 [1] Bioconductor                 
##  yaml                     2.2.1      2020-02-01 [3] CRAN (R 4.0.1)               
##  zlibbioc                 1.36.0     2020-10-27 [1] Bioconductor                 
## 
## [1] /home/yalbi/R/x86_64-pc-linux-gnu-library/4.0
## [2] /usr/local/lib/R/site-library
## [3] /usr/lib/R/site-library
## [4] /usr/lib/R/library
```

]]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
